### PR TITLE
compatibility of imtool3D with Ubuntu

### PR DESCRIPTION
**Pull request by tanguyduval**

### CODE REVIEW

**Comment by tanguyduval** [comment]
@agahkarakuzu I solved the issue on Ubuntu on this branch. Could you please test.
Concerning the doc, you can find it here: https://qmrlab.readthedocs.io/en/master/gui_usage.html#data-viewer

Also, you said it wasn't very intuitive. Could you please give me your feedback on how it could be improved.

Best

**Comment by tanguyduval** [comment]
@agahkarakuzu sorry I just added a feature to export statistics in a text file. That was not possible before! Now I'm done with this branch 😉

**Comment by agahkarakuzu** [comment]
> Also, you said it wasn't very intuitive. Could you please give me your feedback on how it could be improved.

It was all based on the issues I was running into on Ubuntu. I will test it today and let you now :) 

Thanks!

**Comment by agahkarakuzu** [comment]
@tanguyduval I tried it. Sorry, but I will report more bugs on Ubuntu: 

1) When I have labels of different colors such as (no matter which drawing tool I use) :

image

When I click `Statistics` button, expecting to see results from each ROI with respective colors in the spreadsheet, the following error is thrown:

```diff
- Error using horzcat
- Dimensions of arrays being concatenated are not consistent.

- Error in StatsGUI (line 45)
-    Stats_all{Selected+1} = cat(2,[{'Volume'} fields]',cat(1,{'Volume (pixels)','mean', 'median',
-    'std','min','max','1st quartile', '3rd quartile', 'Interquartile Range (IQR)'},Stats));

- Error in qMRLab>Stats_Callback (line 507)
- StatsGUI(I,Maskall, fields, Color);

- Error in gui_mainfcn (line 95)
-        feval(varargin{:});

- Error in qMRLab (line 37)
-    gui_mainfcn(gui_State, varargin{:});

- Error in
- matlab.graphics.internal.figfile.FigFile/read>@(hObject,eventdata)qMRLab('Stats_Callback',hObject,eventdata,guidata(hObject)) 
- Error while evaluating UIControl Callback.
```

2) Once I use either `smart` or `default brush` tool, the default cursor cannot be recovered. Instead, it remains as a circle. While the cursor is like this, other tools are still functional (I can measure distance, draw an elliptical ROI etc.). Having a `reset to default` option can be useful.

**Comment by tanguyduval** [comment]
@agahkarakuzu thank you for these great feedbacks!
1. Tried on Ubuntu but cannot reproduce this bug... I tried with VFA data just like you did, but no bugs on my side. Please detail when you will have time for this
2. solved in ee3f1985ca977a5ec72470b89ebc24fc45d1f053

**Comment by agahkarakuzu** [comment]
@tanguyduval these are the exact steps I followed:

1) Select `vfa_t1` and `load results`
2) Select `red label` then use `paintbrush`
3) Repeat 2 for `green label` and `yellow label` 
4) Hit `Statistics button` 

I had the exact same problem (1) I reported above at the beginning. I also repeated the same steps for `circular ROI`, and the result was the same. 

Then, I noticed that this problem happens when I use a `FitResults` of `v2.0.8`, which is the one in OSF data. When I fit the data, saved a `FitResults` of `v2.0.14`, then repeat the steps (1,2,3,4) above, it worked:

Screen Shot 2019-04-03 at 9 02 49 AM


Problem (2, previous comment) has been resolved after your commits 🎊.

I need to go to Heart Institute now, hope this will give you more hints about the potential origin of the problem. Wonder if it is going to reproduce on your machine.

**Comment by tanguyduval** [comment]
@agahkarakuzu ok got it finally, Thanks!
now solved in a00fd84

File: External/imtool3D_td/src/StatsGUI.m [hunk 1]
```diff
@@ -39,10 +39,10 @@
               'ColumnName',{'Volume (pixels)','mean', 'median', 'std','min','max','1st quartile', '3rd quartile', 'Interquartile Range (IQR)'},...
               'ColumnFormat',{'numeric','numeric', 'numeric', 'numeric','numeric','numeric','numeric','numeric','numeric'},...
               'ColumnEditable',[false false false false false false false false false false],...
-              'RowName',fields','BackgroundColor',Color(Selected+1,:).*[.4 .4 .4]+1-[.4 .4 .4]);
+              'RowName',fields(:),'BackgroundColor',Color(Selected+1,:).*[.4 .4 .4]+1-[.4 .4 .4]);
```
Comment (agahkarakuzu) [inline External/imtool3D_td/src/StatsGUI.m]: 💯

**Comment by agahkarakuzu** [comment]
@tanguyduval I will give it one last try, and if it is all good I will accept this PR. Thank you!

File: External/imtool3D_td/imtool3D.m [hunk 2]
```diff
@@ -613,7 +613,12 @@
             
             % set Image
             setImage(tool, varargin{:})
-
+            
+            % disable ROI tools if no image processing toolbox
```
Comment (agahkarakuzu) [inline External/imtool3D_td/imtool3D.m]: I was just puzzled about how this section is disabling ROI tools.

Comment (agahkarakuzu) [reply External/imtool3D_td/imtool3D.m]: Other than that, all OK.

Comment (tanguyduval) [reply External/imtool3D_td/imtool3D.m]: You're right, that's just a warning for now. Just didn't have time to push this investigation.
I open an issue

**Comment by mathieuboudreau** [comment]
Just dropping in to say that this new feature is SO COOL! Thanks so much @tanguyduval for integrating this feature into qMRLab, and Tommy & Agah for the diligent reviews!

**Pull request merged**